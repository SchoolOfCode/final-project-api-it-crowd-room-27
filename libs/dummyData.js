export const users = [
  {
    firstName: "Jane",
    lastName: "Wilkins",
    email: "jane.wilkins@gmail.com",
    address: "Main Street, LA22 9BU, Ambleside, United Kingdom",
    isActive: false,
    cloudinary_id: "",
    userBio: "hello I'm Jane, i like food",
  },
  {
    firstName: "Dennis",
    lastName: "Thompson",
    email: "dennis.thompson@gmail.com",
    address: "Newlands Meadow, HP11 2BZ, High Wycombe, United Kingdom",
    isActive: true,
    cloudinary_id: "",
    userBio: "hello I'm Dennis, i like drink",
  },
];

export const items = [
  {
    userId: 1,
    category: "fruit",
    itemName: "apple",
    itemDescription: "i have 5 red apples to give",
    useByDate: "2022-02-22",
    dateAdded: "2022-02-18",
    quantity: 5,
    cloudinary_id: "",
    isReserved: false,
    availablity: true,
    timeslot: "between 5pm-7pm",
  },
  {
    userId: 1,
    category: "vegetable",
    itemName: "carrot",
    itemDescription: "i have 10kg of carrots to give",
    useByDate: "2022-02-22",
    dateAdded: "2022-02-18",
    quantity: 1,
    cloudinary_id: "",
    isReserved: true,
    availability: false,
    timeslot: "12:00pm-2:00pm",
  },
];
