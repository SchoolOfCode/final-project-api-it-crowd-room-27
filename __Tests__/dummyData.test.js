import { users } from "../libs/dummyData.js";

test(`the users data is correct`, () => {
  expect(users).toMatchSnapshot();
  expect(users).toHaveLength(2);
  expect(users.map((user) => user.full_name)).toEqual([
    "Jane Wilkins",
    "Dennis Thompson",
  ]);
});

for (let i = 0; i < users.length; i++) {
  test(`users[${i}] should have properties (first_name, last_name, email, address, is_active, cloudinary_id, avatar, user_bio)`, () => {
    expect(users[i]).toHaveProperty("full_name");
    expect(users[i]).toHaveProperty("email");
    expect(users[i]).toHaveProperty("address");
    expect(users[i]).toHaveProperty("is_active");
    expect(users[i]).toHaveProperty("cloudinary_id");
    expect(users[i]).toHaveProperty("avatar");
    expect(users[i]).toHaveProperty("user_bio");
    //  expect(users[i]).toHaveProperty("interesting") => test case for checking property that shouldn't be there, test failed as expected;
  });
}

test("user data has jane and matches as an object", () => {
  const jane = {
    full_name: "Jane Wilkins",
    email: "jane.wilkins@gmail.com",
    address: "Main Street, LA22 9BU, Ambleside, United Kingdom",
    is_active: true,
    cloudinary_id: "",
    avatar:
      "https://cdn.vox-cdn.com/thumbor/00awoM5IS2kFITs9546UyMSePBY=/0x0:2370x1574/1200x800/filters:focal(996x598:1374x976)/cdn.vox-cdn.com/uploads/chorus_image/image/69715362/Screen_Shot_2020_07_21_at_9.38.25_AM.0.png",
    user_bio: "hello I'm Jane, i like food",
  };
  expect(users[0]).toMatchObject(jane);
});
