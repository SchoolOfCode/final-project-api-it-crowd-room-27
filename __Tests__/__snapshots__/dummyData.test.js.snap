// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the users data is correct 1`] = `
Array [
  Object {
    "address": "Main Street, LA22 9BU, Ambleside, United Kingdom",
    "avatar": "https://cdn.vox-cdn.com/thumbor/00awoM5IS2kFITs9546UyMSePBY=/0x0:2370x1574/1200x800/filters:focal(996x598:1374x976)/cdn.vox-cdn.com/uploads/chorus_image/image/69715362/Screen_Shot_2020_07_21_at_9.38.25_AM.0.png",
    "cloudinary_id": "",
    "email": "jane.wilkins@gmail.com",
    "first_name": "Jane",
    "is_active": false,
    "last_name": "Wilkins",
    "user_bio": "hello I'm Jane, i like food",
  },
  Object {
    "address": "Newlands Meadow, HP11 2BZ, High Wycombe, United Kingdom",
    "avatar": "https://cdn.vox-cdn.com/thumbor/00awoM5IS2kFITs9546UyMSePBY=/0x0:2370x1574/1200x800/filters:focal(996x598:1374x976)/cdn.vox-cdn.com/uploads/chorus_image/image/69715362/Screen_Shot_2020_07_21_at_9.38.25_AM.0.png",
    "cloudinary_id": "",
    "email": "dennis.thompson@gmail.com",
    "first_name": "Dennis",
    "is_active": true,
    "last_name": "Thompson",
    "user_bio": "hello I'm Dennis, i like drink",
  },
]
`;
